### ISSUE TITLE
How do I apply QueryDet to RPN?  

### ISSUE THREAD

**Issue by Oswells** [ChenhongyiYang/QueryDet-PyTorch#14]
My questions are:
1) When queryDet is applied to RPN, is the query head parallel to the ROI operator?
2) It is mentioned in section 3.4 of the paper that: "Thirdly, two-stage methods rely on operations likeRoIAlign [15] or RoIPooling [12] to align the features withthe first stage proposal. Nevertheless, they are not used inour approach since we don’t have boxes output in the coarseprediction.“ So do P3 and P4 not have regression and the network discards RoIAlign and RoIPooling?
 
When QueryDet is applied to RPN, is its network structure like this？
![image](https://user-images.githubusercontent.com/46258221/159158559-00cc1539-aef3-40e4-9108-70093c946ad1.png)

**Comment by ChenhongyiYang** [comment]
Hi Oswells,

Both of your understandings are correct. ^.^

**Comment by Oswells** [comment]
Sorry to bother you again.I find my problem contradictory, so I want to confirm it again. 
When QueryDet is applied to a two-stage network, does the network still need RoiAlign or RoIPooling? If not, what is the meaning of the existence of RPN structure. (We can classify and regress the output of FPN directly, so it becomes a single-stage network without RPN structure.)

**Comment by ChenhongyiYang** [comment]
Hi Oswells,

The RoIAlign and RCNN parts keep unchanged, which means for each RoI its features are pooled using RoIAlign, and classification and regression results are computed by the RCNN that usually includes several MLPs. 

The CSQ is only applied to RPN, which is very similar to the single-stage detectors. CSQ accelerates its inference speed and may affect the number of output RoIs, i.e., fewer RoIs for small objects will be generated if the image does not contain small objects. 

In summary, CSQ modifies the way RoI is computed and may change the RoI number. However, it does not affect how the RoIs are treated in the later stage (RoIAligns and RCNN)

**Comment by Oswells** [comment]
Thank you very much! I got it.
